### PR TITLE
Simplification

### PR DESCRIPTION
**Pull request by AlexeyRaga**
## Changes

- Simplifying `Service` pipeline: no need to deal with `Either KafkaError a` and make sure that a value is propagated upstream. This gets rid of templated batching, but simplifies the template.

- Decoding messages from Avro is simplified: `throwM` when the payload cannot be decoded. No need to deal with `Either` all the time.

- Services like logger or stats client are now created in `Main` instead of obscure places like `Application.hs`. This also allows re-using them in `forkIO` threads if needed.

- `ReaderT AppEnv` is simplified and `AppEnv` lensified. 

- Use latest version of Kafka/Avro stuff.

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -20,26 +24,47 @@ main :: IO ()
...
+        logInfo "Instantiating Schema Registry"
+        sr <- schemaRegistry (kafkaConf ^. schemaRegistryAddress)
+
+        logInfo "Running Kafka Consumer"
+        runConduit $
+          kafkaSourceNoClose consumer (kafkaConf ^. pollTimeoutMs)
+          .| throwLeftSatisfy isFatal                   -- throw any fatal error
+          .| skipNonFatalExcept [isPollTimeout]         -- discard any non-fatal except poll timeouts
+          .| tapRight (Srv.handleStream sr)             -- handle messages (see Service.hs)
+          -- .| batchByOrFlushEither (kafkaConf ^. batchSize) -- Use this if you also want a producer.
```
Comment (Kheldar) [inline app/Main.hs]: Have you made sure that this still works?

File: app/Main.hs [hunk 2]
```diff
@@ -20,26 +24,47 @@ main :: IO ()
...
+          .| everyNSeconds (kafkaConf ^. commitPeriodSec)  -- only commit ever N seconds, so we don't hammer Kafka.
+          .| commitOffsetsSink consumer
+          -- .| flushThenCommitSink consumer producer -- Swap with the above if you want a producer.
```
Comment (Kheldar) [inline app/Main.hs]: Also this

File: app/Main.hs (hunk 1 above)
Comment (AlexeyRaga) [reply app/Main.hs]: I have made sure that it won't work and I stated it in the PR description

File: app/Main.hs (hunk 2 above)
Comment (AlexeyRaga) [reply app/Main.hs]: This is not impacted.

File: app/Main.hs (hunk 1 above)
Comment (AlexeyRaga) [reply app/Main.hs]: That's the point: it depends. In different cases you need different solutions.
One of them would be to batch before tapping into `handleStream` so it receives batches in the first place.
Another would be to update `handleStream` to be a `Conduit` and batch them there.
Third solution would be to batch it is a sink, remember offsets and to commit explicit offsets.
We use different strategies in different projects.

**Pull request merged**